Engine/Module/ModulePhysics.cpp

```diff
@@ -75,6 +77,23 @@ update_status ModulePhysics::Update()
 			world->synchronizeSingleMotionState(collider->body);
 		}
 	}
+
+	for(auto collider_a : colliders)
+	{
+		std::vector<CollisionInformation> collisions;
+		for (auto collider_b : colliders)
+		{
+			if (collider_a != collider_b)
+			{
+				CollisionInformation collision_info = collider_a->DetectCollisionWith(collider_b);
+				if (collision_info.collider)
+				{
+					collisions.push_back(collision_info);
+				}
+			}
+		}
+		App->event_manager->publish(new CollisionEvent(collider_a->owner, collisions));
+	}
```

Comment:
Is this for testing? Because you are comparing all colliders at O(n^2). And some collisions will be repeated right? Because it is comparing a with b and b with a.

Reply:
this is done for each element so if it suscribe to the event system will get all the elementes that object is colliding with

Reply:
maybe in order to optimize this you could discard all the objects that are not mesh_collider and their distance is to far away